tests/python/unittest/test_ndarray.py

```diff
@@ -291,6 +293,59 @@ def test_ndarray_legacy_load():
         assert same(data[i].asnumpy(), legacy_data[i].asnumpy())
 
 
+@with_seed()
+def test_buffer_load():
+    nrepeat = 10
+    for repeat in range(nrepeat):
+        # test load_buffer as list
+        data = []
+        for i in range(10):
+            data.append(random_ndarray(np.random.randint(1, 5)))
+        with tempfile.NamedTemporaryFile(mode='wb', delete = False) as tmp:
+            fname = tmp.name
+            tmp.close() # close file so this works in Windows
```

Comment:
This does not work since the with-statement is auto-closing. It seems like you're better off using 
```with TemporaryDirectory() as temp_dir:``` which you'll get with ```from tempfile import TemporaryDirectory```. In there, just manage the files yourself.

Reply:
TemporaryDirectory() doesn't exists in python 2

tempfile.mkdtemp exists in 2 & 3 but doesn't have a with, still that is probably the easiest so I can go with that?

Reply:
I just wrote a simple class in the end :)